### PR TITLE
MongoDB configuration

### PR DESCRIPTION
**Pull request by ha77an**
relates #41
1- created  index.js file in the database.
2- added Mongoose configuration inside index.js in the database folder.
3- called database folder inside server/app.js.

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -19,6 +20,8 @@ const middlewares = [
 
 app.use(middlewares);
 
+dataBase.on('open', console.log.bind(console, 'mongo database is connecting'));
```
Comment (Asmaathabet) [inline server/app.js]: It seems that ESLINT didn't work in your project ??  
console statment is not allowed so it should have thing like this 
```suggestion
// eslint-disable-next-line no-console
dataBase.on('open', console.log.bind(console, 'mongo database is connecting'));
```

File: server/database/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+let dbURI = '';
+if (process.env.NODE_ENV === 'production') {
+  dbURI = process.env.PRODUCT_URI;
+} else {
+  dbURI = process.env.DEV_URI;
+}
+mongoose
+  .connect(dbURI, { useNewUrlParser: true, useCreateIndex: true })
+  .catch((e) => {
+    console.error('Connection error ', e.message);
```
Comment (Asmaathabet) [inline server/database/index.js]: also here the same thing

File: server/app.js (hunk 1 above)
Comment (ha77an) [reply server/app.js]: ok, I will solve it :+1:

File: package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "server": "cd server && npm run dev",
     "client": "cd client && npm start",
     "test": "echo \"Error: no test specified\"",
-    "start": "node server/index.js",
+    "start": "NODE_ENV=production node server/index.js",
     "dev": "nodemon server/index.js",
```
Comment (mFlaifel) [inline package.json]: i think you should add NODE_Env=production to dev script

File: server/database/index.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const mongoose = require('mongoose');
+require('env2')('config.env');
+
+let dbURI = '';
+if (process.env.NODE_ENV === 'production') {
```
Comment (mFlaifel) [inline server/database/index.js]: you can use if else to add development env 
and create another database for development
it is not recommended to modify production database in development stage

**Pull request merged**